src/manifolds/SymmetricPositiveDefiniteFixedDeterminant.jl

```diff
@@ -0,0 +1,152 @@
+    SymmetricPositiveDefiniteFixedDeterminant{N,D} <: AbstractDecoratorManifold{ℝ}
+
+The manifold of symmetric positive definite matrices of fixed determinant ``d > 0``, i.e.
+
+````math
+\mathcal P_d(n) =
+\bigl\{
+p ∈ ℝ^{n × n} \ \big|\ a^\mathrm{T}pa > 0 \text{ for all } a ∈ ℝ^{n}\backslash\{0\}
+  \text{ and } \det(p) = d
+\bigr\}.
+````
+This manifold is modelled as a submanifold of [`SymmetricPositiveDefinite`](@ref)`(n)`.
+
+The tangent space is modelled the same as for [`SymmetricPositiveDefinite`](@ref)`(n)`
+and consists of all symmetric matrices with zero trace
+```math
+    T_p\mathcal P_d(n) =
+    \bigl\{
+        X \in \mathbb R^{n×n} \big|\ X=X^\mathrm{T} \text{ and } \operatorname{tr}(p) = 0
+    \bigr\},
+```
+since for a constant determinant we require that `0 = D\det(p)[Z] = \det(p)\operatorname{tr}(p^{-1}Z)` for all tangent vectprs ``Z``.
+Additionally we store the tangent vectors as `X=p^{-1}Z`, i.e. symmetric matrices.
+
+# Constructor
+
+    SymmetricPositiveDefiniteFixedDeterminant(n::Int, d::Real=1.0)
+
+generates the manifold $\mathcal P_d(n) \subset \mathcal P(n)$ of determinant ``d``,
+which defaults to 1.
+"""
+struct SymmetricPositiveDefiniteFixedDeterminant{N,TD<:Real} <: AbstractDecoratorManifold{ℝ}
+    d::TD
+end
+
+function SymmetricPositiveDefiniteFixedDeterminant(n::Int, d::F=1.0) where {F<:Real}
+    @assert d > 0 "The determinant has to be positive but was provided as $d."
+    return SymmetricPositiveDefiniteFixedDeterminant{n,F}(d)
+end
+
+function active_traits(f, ::SymmetricPositiveDefiniteFixedDeterminant, args...)
+    return merge_traits(IsEmbeddedSubmanifold())
+end
+
+@doc raw"""
+    check_point(M::SymmetricPositiveDefiniteFixedDeterminant{n}, p; kwargs...)
+
+Check whether `p` is a valid manifold point on the [`SymmetricPositiveDefiniteFixedDeterminant`](@ref)`(n,d)` `M`, i.e.
+whether `p` is a [`SymmetricPositiveDefinite`](@ref) matrix (checked by traits) ofsize `(n,n)`
+
+with determinant ``\det(p) = d``.
+
+The tolerance for the determinant of `p` can be set using `kwargs...`.
+"""
+function check_point(
+    M::SymmetricPositiveDefiniteFixedDeterminant{n},
+    p;
+    kwargs...,
+) where {n}
+    if det(p) != M.d
+        return DomainError(
+            det(p),
+            "The point $(p) does not lie on $M, since it does not have a determinant $(M.d).",
+        )
+    end
+    return nothing
+end
+
+"""
+    check_vector(M::SymmetricPositiveDefiniteFixedDeterminant{n}, p, X; kwargs... )
+
+Check whether `X` is a tangent vector to manifold point `p` on the
+[`SymmetricMatrices`](@ref) `M`, i.e. `X` has to be a symmetric matrix of size `(n,n)`
+and its values have to be from the correct [`AbstractNumbers`](https://juliamanifolds.github.io/ManifoldsBase.jl/stable/types.html#number-system).
+
+The tolerance for the symmetry of `X` can be set using `kwargs...`.
+"""
+function check_vector(M::SymmetricPositiveDefiniteFixedDeterminant, p, X; kwargs...)
+    if !isapprox(tr(X), 0.0; kwargs...)
```

Comment:
```suggestion
function check_vector(M::SymmetricPositiveDefiniteFixedDeterminant, p, X; atol::Real=sqrt(max_eps(X)), kwargs...)
    if !isapprox(tr(X), 0.0; atol=atol, kwargs...)
```
So that by default it's not an exact comparison with `0`.

Reply:
hm, until now we never did that.

Reply:
We have that in a few places, for example `KendallsShapeSpace`. I think we should move in this direction for other manifolds as well.

Reply:
hm until now at least my philosophy was: The default from is approx should be fine enough

Reply:
The default is enough when we don't compare to 0. When we compare to 0, we should IMO have a non-zero atol by default.

Reply:
Hm, then is approx has a strange default for that case. I would prefer to maybe have one PR that changes the default behaviour of Manifolds.jl when we compare approximately vs zero overall? Otherwise this would just get a bit inconsistent throughout the package for now.

Reply:
It's already inconsistent but if you prefer we can leave fixing that for later.